plugins/net_plugin/net_plugin.cpp

```diff
@@ -1989,30 +1988,30 @@ namespace eosio {
    }
    //-----------------------------------------------------------
 
-    sync_manager::sync_manager( uint32_t span, uint32_t sync_peer_limit, size_t max_reversible_blocks, uint32_t min_blocks_distance )
+    sync_manager::sync_manager( uint32_t span, uint32_t sync_peer_limit, uint32_t min_blocks_distance )
       :sync_known_lib_num( 0 )
       ,sync_last_requested_num( 0 )
       ,sync_next_expected_num( 1 )
       ,sync_source()
       ,sync_fetch_span( span )
       ,sync_peer_limit( sync_peer_limit )
-      ,max_reversible_blocks(max_reversible_blocks)
       ,sync_state(in_sync)
       ,min_blocks_distance(min_blocks_distance)
    {
    }
 
    uint32_t sync_manager::active_sync_fetch_span() const {
-      auto fork_db_size = my_impl->chain_plug->chain().fork_db_size();
-      int32_t reversible_remaining = max_reversible_blocks - fork_db_size - 1;
+      int32_t reversible_remaining = my_impl->chain_plug->chain().max_reversible_blocks_allowed();
       if (reversible_remaining <= 0) {
-         fc_wlog(logger, "max-reversible-blocks ${m} exceeded, remaining ${r}, fork_db_size ${fs}",
-                 ("m", max_reversible_blocks)("r", reversible_remaining)("fs", fork_db_size));
+         auto fork_db_size = my_impl->chain_plug->chain().fork_db_size();
+         fc_wlog(logger, "max-reversible-blocks exceeded, remaining ${r}, fork_db_size ${fs}",
+                 ("r", reversible_remaining)("fs", fork_db_size));
```

Comment:
We could still print a negative remaining here. Is that intended?

Alternatively, we could check if `reversible_remaining < 0` and print the exceeded number of blocks as a positive number.

For example:

```cpp
fc_wlog(logger, "max-reversible-blocks exceeded by ${ex}, fork_db_size ${fs}",
                 ("ex", -reversible_remaining)("fs", fork_db_size));
```

Reply:
I thought negative remaining would just indicate the node exceeded the threshold. For a warning like this I figured that would be appropriate. I can change to print out positive.